### PR TITLE
Add live and draft rule-tag join tables

### PR DESCRIPTION
**Pull request by rhystmills**
## What does this change?

Follow-up to https://github.com/guardian/typerighter/pull/330

Adds join tables for tags and rules, modelling the many-to-many relationships between the two. These relationships will have a live and draft state, modelled in separate live and draft tables.

This PR only adds:
- the tables
- classes representing the data in those tables
- tests

It does not add routes for managing tag-rule relationships (e.g. endpoints for adding/removing tags to/from rules) or a UI, which will be added in follow-up PRs in order to keep this one simple.

## How to test

1. Run the application locally (`./script.start`)
2. Run the tests (`sbt test`). Do they all pass?

### CODE REVIEW

**Comment by jonathonherbert** [comment]
Just a general comment, as there's some repetition in this PR – have you explored a way of sharing code between tables? I wonder if it might be possible to encapsulate common functionality in a trait, and leave the things that are not common (table name, syntax) to the inheriting objects?

**Comment by rhystmills** [comment]
> Just a general comment, as there's some repetition in this PR – have you explored a way of sharing code between tables? I wonder if it might be possible to encapsulate common functionality in a trait, and leave the things that are not common (table name, syntax) to the inheriting objects?

I wasn't quite sure how to do it but you have compelled me to figure it out, looking much better now!

**Review by jonathonherbert — APPROVED** [review]
Looks great! One non-blocking comment.

File: apps/rule-manager/app/db/RuleTagDraft.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package db
+
+object RuleTagDraft extends RuleTagCommon {
+  override val columns = Seq("rule_id", "tag_id")
```
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply apps/rule-manager/app/db/RuleTagDraft.scala:4]:
```suggestion
```

**Pull request merged**